# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1499611375 14400
#      Sun Jul 09 10:42:55 2017 -0400
# Node ID 26e19cf8e0d9742aef5ddaeb8228ea29d5250969
# Parent  782946d661c8e7d05c55de9ff66369363027a8ad
use class instead of typedef to preserve deprecated function names

* mach-info.h (class oct_mach_info): Define class with individually
deprecated functions since namespace can't be used as a typedef.

diff --git a/liboctave/system/mach-info.h b/liboctave/system/mach-info.h
--- a/liboctave/system/mach-info.h
+++ b/liboctave/system/mach-info.h
@@ -47,14 +47,48 @@ namespace octave
     float_format string_to_float_format (const std::string&);
 
     std::string float_format_as_string (float_format);
   }
 }
 
 #if defined (OCTAVE_USE_DEPRECATED_FUNCTIONS)
 
-OCTAVE_DEPRECATED (4.2, "use 'octave::mach_info' instead")
-typedef octave::mach_info oct_mach_info;
+class oct_mach_info
+{
+public:
+
+  typedef octave::mach_info::float_format float_format;
+
+  OCTAVE_DEPRECATED (4.4, "use 'octave::mach_info::native_float_format' instead")
+  static float_format native_float_format (void)
+  {
+    return octave::mach_info::native_float_format ();
+  }
+
+  OCTAVE_DEPRECATED (4.4, "use 'octave::mach_info::words_big_endian' instead")
+  static bool words_big_endian (void)
+  {
+    return octave::mach_info::words_big_endian ();
+  }
+
+  OCTAVE_DEPRECATED (4.4, "use 'octave::mach_info::words_little_endian' instead")
+  static bool words_little_endian (void)
+  {
+    return octave::mach_info::words_little_endian ();
+  }
+
+  OCTAVE_DEPRECATED (4.4, "use 'octave::mach_info::string_to_float_format' instead")
+  static float_format string_to_float_format (const std::string& str)
+  {
+    return octave::mach_info::string_to_float_format (str);
+  }
+
+  OCTAVE_DEPRECATED (4.4, "use 'octave::mach_info::float_format_as_string' instead")
+  static std::string float_format_as_string (float_format ff)
+  {
+    return octave::mach_info::float_format_as_string (ff);
+  }
+};
 
 #endif
 
 #endif
